src/replication/log_generator.cpp

```diff
@@ -133,6 +133,12 @@ namespace cubreplication
   void
   log_generator::append_repl_object (replication_object &object)
   {
+    /* Be sure that MVCCID is assigned, since MVCCID of m_stream_entry is needed at apply,
+     * to diferentiate between 2 transactions in same group. Also, prevent the situation
+     * when a transaction having replication objects is not added to group.
+     */
+    logtb_get_current_mvccid (&cubthread::get_entry ());
```

Comment:
I think all DB changes potentially producing replication objects should have a MVCCID, independent of HA mode.
My proposal was to add here (before obtaining an MVCCID), an assert:
assert (MVCCID_IS_VALID (tdes->mvccinfo.id));

However, before we catch all cases, we may have several crashes.

Reply:
The purpose of the patch is to quickly fix MVCCID issue for HA. If we don't have MVCCID, in case of HA, we are in trouble. Especially with complete manager, when the system may hangs on slave.

The disadvantage with the patch is that the behavior on single node (no MVCCID used) may be different than HA (MVCCID used). This is the case of "create serial". However, I think that the execution is not affected by current modification. 

I will add a comment to said that is better to be sure that we have MVCCID before append_repl_object. We can catch all that issues with assert in a separate patch.